OverlayPlugin.Core/Integration/FFXIVCustomLogLines.cs

```diff
@@ -65,22 +68,60 @@ public FFXIVCustomLogLines(TinyIoCContainer container)
                     else
                     {
                         var ID = reservedDataEntry.ID.Value;
+                        var Name = reservedDataEntry.Name;
                         if (registry.ContainsKey(ID))
                         {
                             logger.Log(LogLevel.Error, $"Reserved log line entry already registered ({ID}).");
                             continue;
                         }
+                        if (Name == null)
+                        {
+                            logger.Log(LogLevel.Error, $"Reserved log line entry missing Name property");
+                            continue;
+                        }
                         var Source = reservedDataEntry.Source;
                         var Version = reservedDataEntry.Version.Value;
-                        logger.Log(LogLevel.Debug, $"Reserving log line entry for ID {ID}, Source {Source}, Version {Version}.");
+                        logger.Log(LogLevel.Debug, $"Reserving log line entry for ID {ID}, Source {Source}, Name {Name}, Version {Version}.");
                         registry[ID] = new LogLineRegistryEntry()
                         {
                             ID = ID,
+                            Name = Name,
                             Source = Source,
                             Version = Version,
                         };
                     }
+                }
+                if (registry.ContainsKey(registeredCustomLogLineID))
+                {
+                    var entry = registry[registeredCustomLogLineID];
+                    var Source = entry.Source.Replace("\r", "\\r").Replace("\n", "\\n");
+                    var Name = entry.Name.Replace("\r", "\\r").Replace("\n", "\\n");
+                    repository.WriteLogLineImpl(registeredCustomLogLineID, $"{registeredCustomLogLineID}|{Source}|{Name}|{entry.Version}");
                 }
+
+                ActGlobals.oFormActMain.BeforeLogLineRead += (isImport, logInfo) => {

```

Comment:
Do you think it makes sense to merge this now and revert+rerelease if Ravahn updates the plugin to do this directly? Or should we wait for that and remove this now?

Reply:
Maybe it should check for a properly formatted line and handle that instead (basically option 3 from our Discord DM discussion, falling back to option 1 if unformatted)?

Reply:
I guess it depends on what your view of the long term is here.

Personally, I'd rather not have OverlayPlugin be in the business of rewriting parsed log lines (extra work, could get it wrong, seems like incorrect separation of concerns, what if people start depending on it to write triggers).  So, my personal opinion is that we should "fall back to 1 if unformatted" in the short term and if the plugin starts handling it then we can remove the code entirely.

Reply:
The latest beta of the ffxiv plugin looks like it parses these lines now, so I think this could be removed.